toolchest_client/api/download.py

```diff
@@ -58,6 +57,13 @@ def download(output_path, s3_uri=None, pipeline_segment_instance_id=None,
             error_message = "S3 URI of output not provided."
             raise ToolchestDownloadError(error_message) from None
 
+    # Create output directories if output_path does not exist.
+    # Note: Assumes that output_path is a directory if it does not exist.
+    # This may lead to undesired leaf dir creation if output_path is not intended to be a dir,
```

Comment:
What do you think about a warning / error if the leaf of the path has an extension?

I know that would block some valid paths (e.g. test/actually.a.dir), but would probably make usage more clear

Reply:
I'll add a warning message. (This will only be invoked by manual downloads since directory creation in tool runs is handled before downloading, so the warning message won't get in the way of any tool run output.)